subsys/nfc/rpc/server/nfc_rpc_t2t_server.c

```diff
@@ -0,0 +1,229 @@
+ */
+
+#include <nrf_rpc/nrf_rpc_cbkproxy.h>
+#include <nrf_rpc/nrf_rpc_serialize.h>
+
+#include <nfc_rpc_ids.h>
+#include <nfc_rpc_common.h>
+
+#include <nfc_t2t_lib.h>
+
+static void nfc_t2t_cb(void *context, nfc_t2t_event_t event, const uint8_t *data,
+		       size_t data_length, uint32_t callback_slot)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = data ? data_length + 2 : 1;
+
+	cbor_buffer_size += 15; /* for event, context and callback slot */
+	NRF_RPC_CBOR_ALLOC(&nfc_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)context);
+	nrf_rpc_encode_uint(&ctx, event);
+
+	if (data) {
+		nrf_rpc_encode_buffer(&ctx, data, data_length);
+	} else {
+		nrf_rpc_encode_null(&ctx);
+	}
+
+	nrf_rpc_encode_uint(&ctx, callback_slot);
+
+	nrf_rpc_cbor_cmd_no_err(&nfc_group, NFC_RPC_CMD_T2T_CB, &ctx, nrf_rpc_rsp_decode_void,
+				NULL);
+}
+
+NRF_RPC_CBKPROXY_HANDLER(nfc_t2t_cb_encoder, nfc_t2t_cb,
+			 (void *context, nfc_t2t_event_t event, const uint8_t *data,
+			  size_t data_length),
+			 (context, event, data, data_length));
+
+static void nfc_rpc_t2t_setup_rpc_handler(const struct nrf_rpc_group *group,
+					  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	nfc_t2t_callback_t cb;
+	void *cb_ctx;
+	int error;
+
+	cb = (nfc_t2t_callback_t)nrf_rpc_decode_callbackd(ctx, nfc_t2t_cb_encoder);
+	cb_ctx = (void *)nrf_rpc_decode_uint(ctx);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_SETUP);
+		return;
+	}
+
+	error = nfc_t2t_setup(cb, cb_ctx);
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(nfc_group, nfc_rpc_t2t_setup, NFC_RPC_CMD_T2T_SETUP,
+			 nfc_rpc_t2t_setup_rpc_handler, NULL);
+
+static void nfc_rpc_t2t_parameter_set_rpc_handler(const struct nrf_rpc_group *group,
+						  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	nfc_t2t_param_id_t id;
+	void *data;
+	size_t data_length;
+	int error;
+
+	id = nrf_rpc_decode_uint(ctx);
+	data = (void *)nrf_rpc_decode_buffer_ptr_and_size(ctx, &data_length);
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_PARAMETER_SET);
+		return;
+	}
+
+	error = nfc_t2t_parameter_set(id, data, data_length);
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(nfc_group, nfc_rpc_t2t_parameter_set, NFC_RPC_CMD_T2T_PARAMETER_SET,
+			 nfc_rpc_t2t_parameter_set_rpc_handler, NULL);
+
+static void nfc_rpc_t2t_parameter_get_rpc_handler(const struct nrf_rpc_group *group,
+						  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	struct nrf_rpc_cbor_ctx local_ctx;
+	nfc_t2t_param_id_t id;
+	size_t max_data_length;
+	int error;
+
+	id = nrf_rpc_decode_uint(ctx);
+	max_data_length = nrf_rpc_decode_uint(ctx);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_PARAMETER_GET);
+		return;
+	}
+
+	uint8_t data[max_data_length];
```

Comment:
VLA in Zephyr is forbidden (Rule 18.8): https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html. Do you know if this is acceptable in NCS?